File: app/src/main/java/com/pcsalt/example/customalertdialogdemo2/MainActivity.java
package com.pcsalt.example.customalertdialogdemo2;

import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_show_dialog).setOnClickListener(view -> showCustomDialog());
    }

    private void showCustomDialog() {
        LinearLayout ll_Main = new LinearLayout(this);
        LinearLayout ll_Row1 = new LinearLayout(this);
        LinearLayout ll_Row2 = new LinearLayout(this);

        ll_Main.setOrientation(LinearLayout.VERTICAL);
        ll_Main.setPadding(10, 10, 10, 10);
        ll_Row1.setOrientation(LinearLayout.HORIZONTAL);
        ll_Row1.setPadding(10, 10, 10, 10);
        ll_Row2.setOrientation(LinearLayout.HORIZONTAL);
        ll_Row2.setPadding(10, 10, 10, 10);

        final EditText et_User = new MyEditText(this);
        final EditText et_Pass = new MyEditText(this);

        TextView tv_User = new TextView(this);
        TextView tv_Pass = new TextView(this);

        tv_User.setText("Username: ");
        tv_Pass.setText("Password: ");

        ll_Row1.addView(tv_User);
        ll_Row1.addView(et_User);
        ll_Row2.addView(tv_Pass);
        ll_Row2.addView(et_Pass);
        ll_Main.addView(ll_Row1);
        ll_Main.addView(ll_Row2);

        AlertDialog.Builder alert = new AlertDialog.Builder(this);
        alert.setTitle("Custom Login");
        alert.setView(ll_Main);
        alert.setCancelable(false);
        alert.setPositiveButton("Login", (dialog, which) -> {
            String user = et_User.getText().toString();
            String pass = et_Pass.getText().toString();
            Toast.makeText(getBaseContext(), "User: " + user + " Password: " + pass, Toast.LENGTH_SHORT).show();
        });
        alert.setNegativeButton("Cancel", (dialog, which) -> dialog.dismiss());

        AlertDialog dialog = alert.create();
        dialog.show();
    }
}

File: app/src/main/java/com/pcsalt/example/customalertdialogdemo2/MainActivity.kt
package com.pcsalt.example.customalertdialogdemo2

import android.os.Bundle
import android.text.InputType
import android.text.method.PasswordTransformationMethod
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.pcsalt.example.customalertdialogdemo2.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

  private lateinit var binding: ActivityMainBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    binding = ActivityMainBinding.inflate(layoutInflater)
    setContentView(binding.root)
    binding.btnShowDialog.setOnClickListener { showCustomDialog() }
  }

  private fun showCustomDialog() {
    val layout = LinearLayout(this).apply {
      orientation = LinearLayout.VERTICAL
      setPadding(48, 32, 48, 16)
    }

    val etUsername = EditText(this).apply {
      hint = "Username"
      inputType = InputType.TYPE_CLASS_TEXT
      isSingleLine = true
    }

    val etPassword = EditText(this).apply {
      hint = "Password"
      inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
      transformationMethod = PasswordTransformationMethod.getInstance()
      isSingleLine = true
    }

    layout.addView(etUsername)
    layout.addView(etPassword)

    AlertDialog.Builder(this)
      .setTitle("Login")
      .setView(layout)
      .setCancelable(false)
      .setPositiveButton("Login") { _, _ ->
        val user = etUsername.text.toString()
        val pass = etPassword.text.toString()
        Toast.makeText(this, "User: $user Password: $pass", Toast.LENGTH_SHORT).show()
      }
      .setNegativeButton("Cancel") { dialog, _ -> dialog.dismiss() }
      .show()
  }
}
